implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/PrometheusMeterRegistry.java

```diff
@@ -95,28 +107,46 @@ public String scrape() {
     }
 
     /**
-     * Scrape to the specified writer.
+     * Scrape to the specified writer in Prometheus text format.
      *
      * @param writer Target that serves the content to be scraped by Prometheus.
      * @throws IOException if writing fails
      * @since 1.2.0
      */
     public void scrape(Writer writer) throws IOException {
-        TextFormat.write004(writer, registry.metricFamilySamples());
+        scrape(writer, TextFormat.CONTENT_TYPE_004);
+    }
+
+    /**
+     * Write the metrics scrape body in a specific content type to the given writer.
+     *
+     * @param writer where to write the scrape body
+     * @param contentType of the scrape
+     * @throws IOException if writing fails
+     * @see TextFormat
+     * @since 1.7.0
+     */
+    public void scrape(Writer writer, String contentType) throws IOException {
+        scrape(writer, contentType, registry.metricFamilySamples());
+    }
+
+    private void scrape(Writer writer, String contentType, Enumeration<Collector.MetricFamilySamples> samples) throws IOException {
+        TextFormat.writeFormat(contentType, writer, samples);
     }
 
     /**
      * Return text for scraping.
      *
+     * @param contentType the Content-Type of the scrape.
      * @param includedNames Sample names to be included. All samples will be included if {@code null}.
      * @return Content that should be included in the response body for an endpoint designated for
      * Prometheus to scrape from.
-     * @since 1.6.0
+     * @since 1.7.0
      */
-    public String scrape(Set<String> includedNames) {
+    public String scrape(String contentType, @Nullable Set<String> includedNames) {
```

Comment:
I am considering removing these convenience methods since they haven't been released yet and they haven't been requested by any user so far and we have quite a few `scrape` variants now. Spring Boot won't use them since it makes the scrape endpoint from the underlying `CollectorRegistry` instead of our `PrometheusMeterRegistry`. This leaves the only known consumer our sample, which is essentially the Spring Boot Actuator Prometheus endpoint if it used `PrometheusMeterRegistry` (but as stated, it doesn't).

Reply:
I suppose it would be nice to offer the methods for other frameworks to achieve something similar to Spring Boot, but if they also use the underlying CollectorRegistry, then these methods still won't be used.